Formula/trunk.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# Homebrew formula for Trunk https://github.com/thedodd/trunk
+class Trunk < Formula
+  desc "Build, bundle & ship your Rust WASM application to the web"
+  homepage "https://github.com/thedodd/trunk"
+  url "https://github.com/thedodd/trunk/archive/v0.6.0.tar.gz"
+  sha256 "5492729a6dd027c46d91bacca9b3b87beb3b534c7c6f0f2529f9675600a9e273"
+  license any_of: ["MIT", "Apache-2.0"]
+  head "https://github.com/thedodd/trunk.git"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    assert_equal version.to_s, shell_output("#{bin}/trunk --version | tr -d 'trunk ' | tr -d '\n'")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Ok, I've updated this to test `trunk config show` which will process the runtime config of the environment Trunk is running in. This command causes the app to perform a full runtime loop so it should be sufficient based on the requirements you are describing above.